segmentstore/server/src/main/java/io/pravega/segmentstore/server/ContainerEventProcessor.java

```diff
@@ -0,0 +1,118 @@
+     * to its own internal Segment.
+     */
+    CompletableFuture<EventProcessor> forConsumer(@NonNull String name,
+                                                  @NonNull Function<List<BufferView>, CompletableFuture<Void>> handler,
+                                                  @NonNull EventProcessorConfig config);
+
+    /**
+     * Instantiates a new {@link EventProcessor} that only enables to add new data to the internal Segment. This may be
+     * useful when a user needs to store events, but it is not clear how the best approach to process them will be. This
+     * append-only {@link EventProcessor} can be replaced in the future by a regular one with a handler function that
+     * will consume all the events in the internal Segment. If the internal Segment exists, the {@link EventProcessor}
+     * will re-use it. If not, a new internal Segment will be  created. Multiple calls to this method for the same name
+     * should result in returning the same {@link EventProcessor} object.
+     *
+     * @param name     Name of the {@link EventProcessor} object.
+     * @return A {@link CompletableFuture} that, when completed, returns a new {@link EventProcessor} object associated
+     * to its own internal Segment.
+     */
+    CompletableFuture<EventProcessor> forDurableQueue(@NonNull String name);
+
+    /**
+     * The {@link EventProcessor} is a service that takes care of adding and processing events from a durable queue. It
+     * is expected to be unique within a {@link ContainerEventProcessor} (e.g., identified by name), allow callers to
+     * add new elements to the durable queue, and perform ordered, offline processing of events by invoking the handler.
+     * The processing limits of any implementation of this class should be guided by {@link EventProcessorConfig}.
+     */
+    interface EventProcessor extends AutoCloseable {
+
+        /**
+         * Persist a new event to the {@link EventProcessor}'s durable queue. Once the return future completes, it
+         * returns the amount of outstanding byes for this {@link EventProcessor} object.
+         *
+         * @param event     Event to be added to the {@link EventProcessor}'s durable queue.
+         * @param timeout   Maximum amount of time for this operation to be completed.
+         * @return          A {@link CompletableFuture} that, when completed, will acknowledge that the event has been
+         *                  durably stored and returns the amount of outstanding bytes for this {@link EventProcessor}.
+         * @throws TooManyOutstandingBytesException if the {@link EventProcessor} has reached the maximum configured
+         * outstanding bytes.
+         */
+        CompletableFuture<Long> add(@NonNull BufferView event, Duration timeout) throws TooManyOutstandingBytesException;
+    }
+
+    /**
+     * This class provides configuration settings for {@link EventProcessor}.
+     */
+    @Data
+    class EventProcessorConfig {
```

Comment:
As discussed, please make these non-nested.

Reply:
If I remember correctly, we did not agreed on removing these classes from `ContainerEventProcessor`. The comments I remember were:
- Remove code related to `EventProcessor` to `EventProcessorImpl` (Done).
- Make all nested classes `static` (Done). Note that all nested classes in an interface are `static` by default, so I have not defined them as `static class` as that is redundant.

These 2 classes are really linked to `ContainerEventProcessor` (they appear in the API definition itself) and super small, so I do not believe that they deserve to be moved to independent files.